### DISCUSSION TITLE
Hello. What does "Hearbeat" mean? What is the difference between Heartbeat IPC and Cumulative IPC?

### DISCUSSION THREAD

**Discussion by szuhzw** [CMU-SAFARI/Hermes#31] [Q&A] [answered]
![image](https://github.com/CMU-SAFARI/Hermes/assets/65290731/7bd76d04-3f15-4c08-b629-1a9383ecb6e9)

**Comment by szuhzw** [comment]
@rahulbera

**Comment by rahulbera** [comment, marked as answer]
Enabling heartbeat prints a progress message every N instructions. The heartbeat IPC signifies the IPC of the program only since the last heartbeat, whereas the cumulative IPC signifies the IPC since the start of the simulation window.

These questions are very primitive to understand if you follow/read the code.
